File: 2/prime100.py
#소수판단
def is_prime(n):
  for i in range(2, n):
    if n % i == 0:
      return False
  return True

#count 만큰 소수 생성
def get_primes(count):
  res = []
  i = 2
  while len(res) < count:
    if is_prime(i):
      res.append(i)
    i += 1
  return res

print(get_primes(100))

File: 2/prime100.rs
//소수 판단
fn is_prime(n:usize) -> bool {
  for i in 2..n {
    if n % i == 0{
      return false;
    }
  }
  true
}

// 갯수만큼
fn get_primes(n:usize) -> Vec<usize> {
  let mut res = vec![];

    let mut i=2;
    
    loop {
        if is_prime(i) {
            res.push(i);
        }
        i += 1;
        if res.len() == n {
            break;
        }
    }

  res
}

fn main() {
  println!("{:?}", get_primes(100));
}
